codegen/templates/chart/operator-deployment.yamltmpl

```diff
@@ -60,14 +61,15 @@ spec:
       [[- range $key, $value := $customPodLabels ]]
       [[ $key ]]: [[ $value ]]
       [[- end ]]
-[[- if $deploymentStrategy ]]
+[[- if $strategy ]]
   strategy:
-[[- if $deploymentStrategy.Type ]]
-    type: [[ $deploymentStrategy.Type ]]
-[[- end ]]
-[[- if $deploymentStrategy.RollingUpdate ]]
-    rollingUpdate:
-[[ toYaml $deploymentStrategy.RollingUpdate | indent 6 ]]
+[[ toYaml $strategy | indent 4 ]]
+[[- else if $conditionalStrategy ]]
+  strategy:
+[[- range $s := $conditionalStrategy ]]
+{{- if [[ $s.Condition ]] }}
+[[ toYaml $s.Strategy | indent 4 ]]
+{{- end }}
```

Comment:
This seems like a bit much right? We just want to toggle the value of strategy here right? Couldn't we achieve that will something like this:

```yaml
strategy: {{ <user-value> (e.g: Recreate) | default <value> (e.g: RollingUpdate) }}
```

Reply:
We want the strategy value to be automatically set based on a helm condition (if persistence is enabled or not), not based on user input. Something like this:

```yaml
spec:
  strategy:
{{- if $.Values.redis.deployment.persistence.enabled }}
    type: RollingUpdate
{{- end }}
{{- if not $.Values.redis.deployment.persistence.enabled }}
    type: Recreate
{{- end }}
```

The deployment strategy is an object, not a plain string https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#deploymentstrategy-v1-apps That's why we need the `toYaml` function

Reply:
This just seems a bit weird having both in the template. Could we not include the default case since kubernetes would handle that?

Reply:
Actually don't want to block you on this since its pretty small. But I'm good with the changes here. I'll give @EItanya the rest of today to approve otherwise I'll do it